#162170681 Add follow and unfollow functionality

Johnsonojo opened the pull request:
#### What does this PR do?
It has follow and unfollow functionality


#### Description of Task to be completed?
- create follow and unfollow actions
- create follow and unfollow reducers
- write tests for functionality

#### How should this be manually tested?
```bash
git clone git@github.com:andela/balder-ah-frontend.git

cd balder-ah-frontend

npm install

npm run start:dev
```

Visit `/profiles/:username` to follow or unfollow the author.


#### Any background context you want to provide?
A user should be able to follow other authors
A user should be able to unfollow other authors if they choose to


#### What are the relevant pivotal tracker stories? (If applicable)
[#162170681](https://www.pivotaltracker.com/story/show/162170681)


#### Screenshots (If applicable)
<img width="1268" alt="foolllow 1" src="https://user-images.githubusercontent.com/32887459/52403565-466e4100-2ac7-11e9-9d3f-4934bd2c0a57.png">
<img width="1273" alt="foolllow 2" src="https://user-images.githubusercontent.com/32887459/52403566-4706d780-2ac7-11e9-9a68-395e4acc5e19.png">




#### Questions:
N/A

akhilome commented on src/App.jsx:
```diff
@@ -39,6 +40,8 @@ class App extends Component {
             <Route path="/articles" component={AllArticles} />
             <PrivateRoute path="/profile" component={ProfilePage} exact />
             <PrivateRoute path="/profile/articles" component={AllUserArticles} exact />
+            <Route path="/profile" exact component={ProfilePage} />
```
Why this again. It has already been defined as a `PrivateRoute` on line 41

Johnsonojo replied:
Thanks for spotting that. Will fix it

akhilome commented on src/__tests__/actions/reactions/follow.js:
```diff
@@ -0,0 +1,91 @@
+      await axiosMock.onPost().replyOnce(403, payload);
+      await getUserFollowers(username)(dispatch);
+      expect(dispatch).toHaveBeenCalledTimes(1);
+      //   expect(dispatch).toHaveBeenCalledWith({
```
Commented out code

Johnsonojo replied:
Taken care of

akhilome commented on src/components/profile/FollowAuthor.jsx:
```diff
@@ -0,0 +1,137 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+// import { withRouter } from 'react-router-dom';
```
Commented out chunk

Johnsonojo replied:
Resolved thanks

akhilome reviewed (commented):
Good job with this @Johnsonojo, I left some comments below 👇

Johnsonojo committed:
[Feature #162170681] Add follow and unfollow functionality

- create follow and unfollow actions
- create follow and unfollow reducers
- wtite test for functionality

[Finishes #162170681]

oluwajuwon commented on src/reducers/profile.js:
```diff
@@ -38,6 +40,15 @@ export default (state = initialState, action) => {
+    case type.GET_OTHER_PROFILE:
+      return {
+        ...state,
+        username: action.payload.userProfile.username,
```
Nice work sof ar @Johnsonojo, wouldn't you rather destructure here?

oluwajuwon reviewed (commented):
Nice job @Johnsonojo 👏🏿

I left a comment on your PR please have a look
